secretstores/hashicorp/vault/vault.go

```diff
@@ -112,9 +114,10 @@ type vaultListKVResponse struct {
 // NewHashiCorpVaultSecretStore returns a new HashiCorp Vault secret store.
 func NewHashiCorpVaultSecretStore(logger logger.Logger) secretstores.SecretStore {
 	return &vaultSecretStore{
-		client: &http.Client{},
-		logger: logger,
-		json:   jsoniter.ConfigFastest,
+		client:   &http.Client{},
+		logger:   logger,
+		features: []secretstores.Feature{secretstores.FeatureMultipleKeyValuesPerSecret},
```

Comment:
is there a reason to keep the features as a property? I mean, if not, I think you can return straight the feature array on line 512

Reply:
Not really. I thought this would be more aligned to what I saw on other state-store components but I guess simpler is better here, I will update the code.